dll/ntdll/ldr/ldrapi.c

```diff
@@ -1672,4 +1672,23 @@ LdrSetAppCompatDllRedirectionCallback(
     return STATUS_NOT_IMPLEMENTED;
 }
 
+BOOLEAN
+NTAPI
+LdrInitShimEngineDynamic(IN PVOID BaseAddress)
+{
+    ULONG_PTR Cookie;
+    NTSTATUS Status = LdrLockLoaderLock(0, NULL, &Cookie);
+    if (NT_SUCCESS(Status))
+    {
+        if (!g_pShimEngineModule)
+        {
+            g_pShimEngineModule = BaseAddress;
+            LdrpGetShimEngineInterface();
+        }
+        LdrUnlockLoaderLock(0, Cookie);
+        return TRUE;
+    }
+    return FALSE;
```

Comment:
If you return FALSE early here, you can reduce indentation level :smile:

Reply:
But why?

Reply:
Well if there's nothing to be done additionaly in a failure case, this would simplify the code. Simpler is usually better, not always though, but in this case why not.

Reply:
Introducing multiple exit points does not make the code simpler imo.